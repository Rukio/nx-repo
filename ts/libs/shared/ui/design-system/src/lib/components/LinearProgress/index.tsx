import MuiLinearProgress, {
  LinearProgressProps as MuiLinearProgressProps,
} from '@mui/material/LinearProgress';
import Typography, { TypographyProps } from '../Typography';

const Title = (props: TypographyProps) => {
  const { children, ...rest } = props;

  return (
    <Typography
      sx={{
        fontSize: '12px',
        mb: 1,
      }}
      {...rest}
    >
      {children}
    </Typography>
  );
};

const TitleUppercase = (props: TypographyProps) => {
  const { children, ...rest } = props;

  return (
    <Title textTransform="uppercase" {...rest}>
      {children}
    </Title>
  );
};

export interface LinearProgressProps extends MuiLinearProgressProps {
  title?: string;
  titleUppercase?: boolean;
}

const LinearProgress = ({
  title,
  titleUppercase,
  ...rest
}: LinearProgressProps) => {
  const TitleComponent = titleUppercase ? (
    <TitleUppercase>{title}</TitleUppercase>
  ) : (
    <Title>{title}</Title>
  );

  return (
    <>
      {!!title && TitleComponent}
      <MuiLinearProgress {...rest} />
    </>
  );
};

export default LinearProgress;
