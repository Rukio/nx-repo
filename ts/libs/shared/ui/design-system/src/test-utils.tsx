import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider, theme } from './lib';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
  userEventOptions?: UserEventSetupConfig;
};

const WrappedProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: CustomRenderOptions
) => {
  return {
    user: userEvent.setup(options?.userEventOptions),
    ...render(ui, { wrapper: WrappedProviders, ...options?.renderOptions }),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
