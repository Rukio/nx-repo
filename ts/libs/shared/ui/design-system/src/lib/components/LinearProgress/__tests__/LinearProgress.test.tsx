import { render } from '../../../../test-utils';

import { Basic } from '../__storybook__/LinearProgress.stories';

test('should render', () => {
  const { asFragment } = render(<Basic />);
  expect(asFragment()).toMatchSnapshot();
});
