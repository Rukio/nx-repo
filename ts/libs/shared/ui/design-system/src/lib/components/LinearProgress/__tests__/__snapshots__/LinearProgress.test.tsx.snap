// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    left: -35%;
    right: 100%;
  }

  60% {
    left: 100%;
    right: -90%;
  }

  100% {
    left: 100%;
    right: -90%;
  }
}

@keyframes animation-1 {
  0% {
    left: -200%;
    right: 100%;
  }

  60% {
    left: 107%;
    right: -8%;
  }

  100% {
    left: 107%;
    right: -8%;
  }
}

.emotion-0 {
  width: 100%;
}

.emotion-1 {
  position: relative;
  overflow: hidden;
  display: block;
  height: 4px;
  z-index: 0;
  background-color: rgb(158, 202, 228);
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2 {
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  top: 0;
  -webkit-transition: -webkit-transform 0.2s linear;
  transition: transform 0.2s linear;
  transform-origin: left;
  background-color: #0074B8;
  width: auto;
  -webkit-animation: animation-0 2.1s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;
  animation: animation-0 2.1s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;
}

.emotion-3 {
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  top: 0;
  -webkit-transition: -webkit-transform 0.2s linear;
  transition: transform 0.2s linear;
  transform-origin: left;
  background-color: #0074B8;
  width: auto;
  -webkit-animation: animation-1 2.1s cubic-bezier(0.165, 0.84, 0.44, 1) 1.15s infinite;
  animation: animation-1 2.1s cubic-bezier(0.165, 0.84, 0.44, 1) 1.15s infinite;
}

<div
    class="MuiBox-root emotion-0"
  >
    <span
      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate emotion-1"
      role="progressbar"
    >
      <span
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate emotion-2"
      />
      <span
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate emotion-3"
      />
    </span>
  </div>
</DocumentFragment>
`;
